### DISCUSSION TITLE
LdapRecord for ActiveDirectory getting error

### DISCUSSION THREAD

**Discussion by terabull007** [Laravel-Backpack/community-forum#799] [Q&A (Help)]
Trying to integrate ldapRecord plugin for ActiveDirectory, I am able to connect to the AD using ldapRecord and get the list of users, but getting error page while all the configurations seem to be correct.
I have tried all available resources, but with no results. with regard to LdapRecord and backpack there are way too less resources available.
I am using Backpack 6 on Windows Server 2019, whatever changes i make, i am getting the below error message
Error: Call to undefined method LdapRecord\Query\Model\ActiveDirectoryBuilder::getTable()

Thanks in advance

**Comment by pxpm** [comment]
@terabull007 I've never used AD login nor I ever used WS, so my knowledge here may be a little limited.

There is a function in the Laravel Models that's `getTable()`. I think part of the implementation of that package is replacing the Laravel User Model with their own model or something ? 

Basically what it says is that somewhere we are trying to call `$model->getTable()`, but instead of an `App\Model` we have that `LdapRecord\Query\Model` that does not have the `getTable()` function 🤷‍♂️ 

Sorry can't help much more, the error you provided does not give much more info. 

Cheers

**Reply by terabull007** [reply]
thank you pxpm, your understanding is correct, even i was thinking in the same lines
I am able to connect to AD with _php artisan ldap:test_ command
I am also able to import users to local database using _php artisan ldap:import users_
But finding difficult to configure backpack login to LdapRecord

**Reply by pxpm** [reply]
It's very likely that you need to implement your own LoginController and probably another authentication middleware no sure. 

Does the package have instructions on how to setup the auth stuff (login, logout etc) ? 

You should be able to do exctly the same, you can just go to backpack config, disable backpack auth routes etc, and follow the package tutorial on how to create the login. 

Make sure you also configure the `backpack_guard` to the one you use when creating the auth mechanism. By default Laravel brings `web`, and backpack brings `backpack`. 

Cheers

**Comment by ale1981** [comment]
@terabull007 can you show your config/auth.php file?

I have Backpack working fine with LdapRecord using Windows AD.

**Comment by jcastroa87** [comment]
Due to not activity i will close this issue, but please feel free to re-open or create a new one if needed.

Cheers.
